vfs-azure/src/main/java/com/sludev/commons/vfs2/provider/azure/AzFileObject.java

```diff
@@ -512,136 +492,182 @@ protected boolean doSetLastModifiedTime(long modtime) throws Exception {
-                            else {
-                                log.debug("currBlobProperties is null");
-                            }
-
-                            OperationContext opContext = new OperationContext();
-                            opContext.setLoggingEnabled(ENABLE_AZURE_STORAGE_LOG);
-
-                            fileCurrBlob.upload(sourceStream, length, null, null, opContext);
-
-                        }
-                        finally {
-                            destFile.close();
-                            srcFile.close();
-                        }
+                        doCopyFromStream(srcFile, destFile);
                     }
                     else {
                         // nothing useful to do if no content and can't have children
                         throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile },
                                 new UnsupportedOperationException());
                     }
                 }
-                catch (IOException io) {
-                    throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile }, io);
+                catch (IOException | BlobStorageException e) {
+                    throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile }, e);
                 }
-                catch (StorageException se) {
-                    throw new FileSystemException("vfs.provider/copy-file.error", new Object[] { srcFile, destFile }, se);
+            }
+        }
+        catch (Exception e) {
+            throw new FileSystemException(e);
+        }
+    }
+
+
+    /**
+     * Copy a file object via input/output streams.
+     *
+     * @param srcFile
+     * @param destFile
+     * @throws IOException
+     */
+    private void doCopyFromStream(FileObject srcFile, FileObject destFile) throws Exception {
+
+        try {
+            String destFilename = destFile.getName().getPath();
+
+            if (destFilename.startsWith("/")) {
+                destFilename = destFilename.substring(1);
+            }
+
+            BlobClient destBlobClient = blobContainerClient.getBlobClient(destFilename);
+
+//                            int blockSize = getBlockSize(srcFile.getContent().getSize(), UPLOAD_BLOCK_SIZE);
+
+            int blockSize = 4 * 1024 *1024;
+
+            ParallelTransferOptions opts =
+                    new ParallelTransferOptions(blockSize, 5, null);
+
+            BlobRequestConditions requestConditions = new BlobRequestConditions(); //.setIfNoneMatch(Constants.HeaderConstants
+            // .ETAG_WILDCARD);
+
+
+            BlobOutputStream bos = destBlobClient.getBlockBlobClient().getBlobOutputStream(
+                    opts, null, null, null, requestConditions);
+
+            InputStream is = srcFile.getContent().getInputStream();
+
+            try {
+                byte[] buffer = new byte[4 * 1024 *1024];
```

Comment:
looks like a repeat of the magic number defined above into `blockSize` - we should make this reference the same value, whether that remains hardcoded or comes from configuration

Reply:
This one does not come from configuration. The fixed size is intentional.

Reply:
OK, in which case can we perhaps extract this to a private static final constant with a name explaining why its value is what it is?

(also perhaps in initialising this constant we can make use of the existing MB multiplier constant instead of `* 1024 * 1024`)